horovod/common/gradient_aggregation.py

```diff
@@ -126,18 +163,27 @@ def compute_gradients(self, grads):
 
         with tf.control_dependencies([update_counter]):
             if self.aggregation_frequency > 1:
+                grads = get_not_none_from_list(grads)
+                assert len(grads) == len(self.gpu_shadow_vars)
                 allreduced_grads = tf.cond(
                     tf.equal(self.counter, self.aggregation_frequency),
                     lambda: self._allreduce_grads_helper(grads),
                     lambda: grads,
                 )
+                assert len(allreduced_grads) == len(self.gpu_shadow_vars)
+                for idx in self.none_indexes:
+                    allreduced_grads.insert(idx, None)
```

Comment:
I think it would be better if we did. This way we aren't going around inserting and resizing a list a bunch of times. Also Python list comprehensions are _fast_:

```
new_allreduced_grads = [grad if idx not in set(self.none_indexes) else None for idx,grad in enumerate(allreduced_grads)]
```

Reply:
Had to do this slightly differently but its done and tested.